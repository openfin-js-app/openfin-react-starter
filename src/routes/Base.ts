import * as React from 'react';
import {SvgIconProps} from '@material-ui/core/SvgIcon'

export type RouteItem = (RouteCompItem|RouteRedirectItem);

export interface RouteCompItem {
    path:string,
    sidebarName?:string,
    navbarName?:string,
    icon?:React.ComponentType<SvgIconProps>,
    component:React.ComponentType<any>,
    [key:string]:any,
}

export interface RouteRedirectItem {
    redirect:boolean,
    path:string,
    to:string,
    navbarName:string,
}