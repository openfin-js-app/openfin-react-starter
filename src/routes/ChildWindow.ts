import AccessibilityIcon from '@material-ui/icons/Accessibility';
import BusinessIcon from '@material-ui/icons/Business';
import ChatIcon from '@material-ui/icons/Chat';
import ReportIcon from '@material-ui/icons/Report';

import { RouteItem } from './Base';

import Accessibility from '../views/Accessibility/Accessibility';
import ViewOne from '../views/ViewOne/ViewOne';
import ViewTwo from '../views/ViewTwo/ViewTwo';
import ReportView from '../views/ReportView/ReportView';

const childRoutes:RouteItem[]=[
    {
        path:'/childWindow/accessibility',
        sidebarName:"Accessibility",
        navbarName:"View accessibility",
        icon: AccessibilityIcon,
        component: Accessibility,
    },
    {
        path:'/childWindow/view-one',
        sidebarName:"ViewOne",
        navbarName:"View one",
        icon: BusinessIcon,
        component: ViewOne,
    },
    {
        path:'/childWindow/view-two',
        sidebarName:"ViewTwo",
        navbarName:"View two",
        icon: ChatIcon,
        component: ViewTwo,
    },
    {
        path:'/childWindow/report',
        sidebarName:"Report",
        navbarName:"Desktop runtime report",
        icon: ReportIcon,
        component: ReportView,
    },
];

export default childRoutes;