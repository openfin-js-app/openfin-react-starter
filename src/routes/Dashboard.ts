import AccessibilityIcon from '@material-ui/icons/Accessibility';
import BusinessIcon from '@material-ui/icons/Business';
import ChatIcon from '@material-ui/icons/Chat';
import ReportIcon from '@material-ui/icons/Report';
import SettingsIcon from '@material-ui/icons/Settings';

import { RouteItem } from './Base';

import Accessibility from '../views/Accessibility/Accessibility';
import ViewOne from '../views/ViewOne/ViewOne';
import ViewTwo from "../views/ViewTwo/ViewTwo";
import ReportView from "../views/ReportView/ReportView";
import ConfigView from '../views/ConfigView/ConfigView';

const dashboardRoutes:RouteItem[]=[
    {
        path:'/dashboard/accessibility',
        sidebarName:"accessibility.sidebar",
        navbarName:"accessibility.navbar",
        icon: AccessibilityIcon,
        component: Accessibility,
    },
    {
        path:'/dashboard/view-one',
        sidebarName:"viewOne.sidebar",
        navbarName:"viewOne.navbar",
        icon: BusinessIcon,
        component: ViewOne,
    },
    {
        path:'/dashboard/view-two',
        sidebarName:"viewTwo.sidebar",
        navbarName:"viewTwo.navbar",
        icon: ChatIcon,
        component: ViewTwo,
    },
    {
        path:'/dashboard/config',
        sidebarName:"configuration.sidebar",
        navbarName:"configuration.navbar",
        icon: SettingsIcon,
        component: ConfigView,
    },
    {
        path:'/dashboard/report',
        sidebarName:"report.sidebar",
        navbarName:"report.navbar",
        icon: ReportIcon,
        component: ReportView,
    },
];

export default dashboardRoutes;