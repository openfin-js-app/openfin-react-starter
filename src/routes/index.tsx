import * as React from 'react';
import {
    RouteItem,
    LoadingLyt,
    DashboardLyt,
    ChildWindowLyt,
    NotificationLyt,
    LaunchBarLyt,
} from 'react-openfin-mat-impl';

import dashboardRoutes from './Dashboard';
import childrenRoutes from './ChildWindow';
import notificationRoutes from './notification';
import launchBarItems from '../constants/launchBarItems';

import appLogo from '../assets/svg/app.svg';
import companyLogo from '../assets/svg/company.svg';

const ENABLE_LOADING_VIEW = process.env.REACT_APP_ENABLE_LOADING_VIEW.toLowerCase() === 'true';

const Loading:React.FunctionComponent<{}> = ({})=>{
    return (<LoadingLyt appLogo={appLogo} companyLogo={companyLogo} version={process.env.REACT_APP_VERSION}/>)
}

const Dashboard:React.FunctionComponent<{}> = ({...rest}) => {
    return (<DashboardLyt appLogo={appLogo} routes={dashboardRoutes} {...rest}/>)
}

const ChildWindow:React.FunctionComponent<{}> = ({...rest}) => {
    return (<ChildWindowLyt appLogo={appLogo} routes={childrenRoutes} {...rest}/>)
}

const Notification:React.FunctionComponent<{}> = ({})=>{
    return (<NotificationLyt routes={notificationRoutes}/>)
}

const LaunchBar:React.FunctionComponent<{}> = ({})=>{
    return (<LaunchBarLyt appLogo={appLogo} items={launchBarItems}/>)
}

const indexRoutes:RouteItem[] = [
    {
        path:'/loading',
        component:Loading,
    },
    {
        path:'/dashboard',
        component: Dashboard,
    },
    {
        path:'/childWindow',
        component: ChildWindow,
    },
    {
        path:'/notification',
        component: Notification,
    },
    {
        path:'/launchBar',
        component: LaunchBar,
    },
    {
        redirect: true, path:'/',
        to: ENABLE_LOADING_VIEW?"/loading":process.env.REACT_APP_DEFAULT_VIEW_URL,
        navbarName:"Redirect"
    }
];

export default indexRoutes;