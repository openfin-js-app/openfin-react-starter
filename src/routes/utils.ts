import {RouteItem} from 'react-openfin-mat-impl';

let localPublicPathname;
try {
    localPublicPathname = new URL(process.env.PUBLIC_URL).pathname
}catch(e){
    localPublicPathname=process.env.PUBLIC_URL
}

export const publicPathname = localPublicPathname;

export function mapToPublicPathname(routeItem:RouteItem):RouteItem {
    routeItem.path = localPublicPathname + routeItem.path;
    return routeItem;
}
