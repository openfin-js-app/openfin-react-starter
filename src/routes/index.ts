import { RouteItem } from './Base';
import Loading from '../layouts/Loading/Loading';
import Dashboard from '../layouts/Dashboard/Dashboard';

export * from './Base';

const ENABLE_LOADING_VIEW = process.env['REACT_APP_ENABLE_LOADING_VIEW'].toLowerCase() === 'true';

const indexRoutes:RouteItem[] = [
    {
        path:'/loading',
        component:Loading,
    },
    {
        path:'/dashboard',
        component: Dashboard,
    },
    {
        redirect: true, path:'/',
        to: ENABLE_LOADING_VIEW?"/loading":process.env['REACT_APP_DEFAULT_VIEW_URL'],
        navbarName:"Redirect"
    }
];

export default indexRoutes;