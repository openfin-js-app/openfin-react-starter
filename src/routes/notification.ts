import { RouteItem } from 'react-openfin-mat-impl';

import SampleNotification from '../views/NotificationViews/SampleNotification';

import { mapToPublicPathname } from './utils';

const notificationRoutes:RouteItem[]=[
    {
        path:'/notification/sample',
        component: SampleNotification,
    }
]

export default notificationRoutes.map(mapToPublicPathname);