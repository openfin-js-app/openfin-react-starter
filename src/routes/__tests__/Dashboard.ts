import routes from '../Dashboard';

describe('Dashboard routes',()=>{

    it('default routes',()=>{
        expect(routes).toMatchSnapshot();
    })

})