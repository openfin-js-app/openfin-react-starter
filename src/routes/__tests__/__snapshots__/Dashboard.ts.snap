// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard routes default routes 1`] = `
Array [
  Object {
    "component": [Function],
    "icon": Object {
      "$$typeof": Symbol(react.memo),
      "compare": null,
      "displayName": "AccessibilityIcon",
      "muiName": "SvgIcon",
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      },
    },
    "navbarName": "accessibility.navbar",
    "path": "/dashboard/accessibility",
    "sidebarName": "accessibility.sidebar",
  },
  Object {
    "component": [Function],
    "icon": Object {
      "$$typeof": Symbol(react.memo),
      "compare": null,
      "displayName": "BusinessIcon",
      "muiName": "SvgIcon",
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      },
    },
    "navbarName": "viewOne.navbar",
    "path": "/dashboard/view-one",
    "sidebarName": "viewOne.sidebar",
  },
  Object {
    "component": [Function],
    "icon": Object {
      "$$typeof": Symbol(react.memo),
      "compare": null,
      "displayName": "ChatIcon",
      "muiName": "SvgIcon",
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      },
    },
    "navbarName": "viewTwo.navbar",
    "path": "/dashboard/view-two",
    "sidebarName": "viewTwo.sidebar",
  },
  Object {
    "component": [Function],
    "icon": Object {
      "$$typeof": Symbol(react.memo),
      "compare": null,
      "displayName": "SettingsIcon",
      "muiName": "SvgIcon",
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      },
    },
    "navbarName": "configuration.navbar",
    "path": "/dashboard/config",
    "sidebarName": "configuration.sidebar",
  },
  Object {
    "component": [Function],
    "icon": Object {
      "$$typeof": Symbol(react.memo),
      "compare": null,
      "displayName": "ReportIcon",
      "muiName": "SvgIcon",
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      },
    },
    "navbarName": "report.navbar",
    "path": "/dashboard/report",
    "sidebarName": "report.sidebar",
  },
]
`;
